content/docs/examples/advanced-gateways/egress-mongo/index.md

```diff
@@ -0,0 +1,1111 @@
+instructions in this section. Alternatively, if you do want to direct your traffic through an egress gateway, proceed to
+[Direct TCP egress traffic through an egress gateway](#direct-tcp-egress-traffic-through-an-egress-gateway).
+
+1.  Define a TCP mesh-external service entry:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: mongo
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      addresses:
+      - $MONGODB_IP/32
+      ports:
+      - number: $MONGODB_PORT
+        name: tcp
+        protocol: TCP
+      location: MESH_EXTERNAL
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application. Now the application should display the ratings without error:
+
+{{< image width="80%" ratio="36.69%"
+    link="./externalDBRatings.png"
+    caption="Book Ratings Displayed Correctly"
+    >}}
+
+Note that you see a one-star rating for both displayed reviews, as expected. You set the ratings to be one star to
+provide yourself with a visual clue that your external database is indeed being used.
+
+#### Cleanup of the egress configuration for TCP without a gateway
+
+{{< text bash >}}
+$ kubectl delete serviceentry mongo
+{{< /text >}}
+
+### Direct TCP Egress traffic through an egress gateway
+
+In this section you handle the case when you need to direct the traffic through an
+[egress gateway](/docs/examples/advanced-gateways/egress-gateway/#use-case). The sidecar proxy routes TCP
+connections from the MongoDB client to the egress gateway, by matching the IP of the MongoDB host (a CIDR block of
+  length 32). The egress gateway forwards the traffic to the MongoDB host, by its hostname.
+
+1.  Create a `ServiceEntry` for the MongoDB service, this time with `resolution` `DNS`. Specifying the resolution as
+    `DNS` instructs the egress gateway to perform a DNS query to get the IP address of the MongoDB host. Note that the
+    IP of the MongoDB host used by the MongoDB client (the `ratings` service) is not known to the egress gateway.
+    The sidecar proxy of the `ratings` service forwards the connection from the `ratings` service to the egress gateway,
+    so the IP of the egress gateway is used as the destination IP.
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: mongo
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      addresses:
+      - $MONGODB_IP/32
+      ports:
+      - number: $MONGODB_PORT
+        name: tcp
+        protocol: TCP
+      resolution: DNS
+      location: MESH_EXTERNAL
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application and verify that the ratings are displayed correctly.
+
+1.  Proceed to the following section.
+
+#### Mutual TLS between the sidecar proxies and the egress gateway
```

Comment:
This seems like the less common, more advanced, case, so I would show it last.

Reply:
Done